### ISSUE TITLE
Linux container using compose volume unusable

### ISSUE THREAD

**Issue by jardon** [docker/for-win#6847]
While running a Linux container in Docker for Windows that uses a volume mount, the containers appear to be working but are incredibly slow regardless of how many resources have been given in Hyper-V.  If I remove the volume and just copy the files over directly, the containers work fine.  I'm assuming that something with how Docker handles filesystems between the host, vm, and containers.  

The setup runs flawlessly on a Linux host and allows for hot reload on my node application.  
  - [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 356525BD-27B8-48B9-8247-BAC1DDAA8368/20200524152329

### Expected behavior
Linux containers should open and run.  Any changes to the mounted codebase should be reflected in the application.

### Actual behavior
The containers start, but on my npm install the containers slow to a crawl and eventually hang.  Exiting the containers usually results in all of docker freezing and needing a reboot.  The volume does mount as I can see all of the files present in the container, but for whatever reason there is very poor performance.

### Information
This is my first time using Docker for Windows cause I usually develop in Linux so I'm not sure if this a new issue or not.  I will continue to use Linux, but other people working on the codebase are using Windows hosts so its a blocker.
  - Windows Version: 10.0.18363 Build 18363
  - Docker Desktop Version: 2.3.0.2 (45183)
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: No

### Steps to reproduce the behavior
1. Clone git@github.com:jardon/ast-training.git
2. Run docker-compose up
3. Wait for it to hang

**Comment by ManfredLange** [comment]
Similar issue here: It seems that anything that has to do with the file system on mounts is either incredibly slow or coming to a grinding halt (hanging) very quickly. This also applies to comands such as `ls -lart` in a bash terminal within the container. The command simply doesn't execute.

Similarly:
- `npm ci` (should remove ./node_modules but doesn't)
- `rm -rf ./node_modules`

One factor could be if the host was a laptop that went to sleep or hibernation over night. Yesterday, deleting the container in question the restarting Docker Desktop helped resolved it. Today, same issue but even deleting the container and restarting Docker Desktop hasn't helped just yet. Perhaps there is yet another factor that is relevant to this issue that I haven't identified yet.

**Comment by abarani** [comment]
Same issue here with `yarn` not being able to read some files and then the container hangs indefinitely. Stopping the container doesn't work, I must restart Docker.

Worth noting that version 2.2.0.5 works fine.

**Comment by mcs07** [comment]
I see the same thing on 2.3.0.3 (45519)

**Comment by nev7n** [comment]
I have the same issue, also on 2.3.03 (45519).  Running commands that make heavy use of mounted filesystem (like `yarn install`) periodically hang then it isn't possible to kill the container without restarting docker desktop.

Diagnostic ID: 
A12740BB-2A81-4635-85CF-727B60B9E1B3/20200619104833

**Comment by pgrefviau** [comment]
Diagnostic ID: 171D2737-3EA3-4D62-AA8B-3040D5E7C69C/20200924210025

Getting what looks to be the same issue on 2.3.0.4. A container set-up to run terraform/terragrunt will consistently hang when running operations that end-up creating a lot of files on a volume (mounted as `-v /work`, no other options). Any docker operations (`kill`, `stop`, `attach`, etc.) operating on the container also hang after this occurs: the only way to stop the container is to restart the docker service.

I also see the same `FS_Q_OVERFLOW detected: flushing all caches` pattern that some other people in this thread also noticed appear before the container hangs. One additional thing that I noticed is that, right _after_ a container's forced shutdown via a docker service restart, there is a high number of `unable to inject event for path` events that appear in the logs, with an error message that's either `no such file or directory` or an RPC error blob with a description that mentions `failed to write client preface: The pipe is being closed.`

The fact that there's usually 50+ of those events, the timing at which they get emitted and the fact that they're (AFAICT) all under the path that is currently being "flushed" leads me me to believe that they're likely to be directly tied to the issue.

I don't want to muddy the water with speculations, but one more thing that I've noticed is that the paths in the  `unable to inject event for path` (and those that I'm dealing with in general as part the container's usage) are pretty long, as in usually longer than 150 characters at a quick glance. Now, while trying to get rid of this show-stopping issue, I installed docker edge releases that actually seemed to be rid of the current issue (except that other, sometimes more critical issues popped-up). As part of one of these games of edge-release Russian roulette, I ran into a situation where some of the mounted directories' inodes ended-up as corrupted within the container. ex:

```
> ls -la
d????????? ? ?    ?       ?            ? plugins
```

These directories not be deleted from within the container, and trying to delete them from the Windows side revealed that the paths were over the max allowed size. I can't even tell what version this was anymore, and I don't think it'll be easy to repro this, but now that I'm back to 2.3.0.4, I'm wondering if file path length might not also play a role in this particular problem as well. Maybe there's a pattern where other people experiencing those hangs are also dealing with long paths? Food for thought.
